src/main/java/com/github/princesslana/eriscasper/gateway/commands/UpdateVoiceState.java

```diff
@@ -0,0 +1,56 @@
+package com.github.princesslana.eriscasper.gateway.commands;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.princesslana.eriscasper.data.Snowflake;
+import com.github.princesslana.eriscasper.gateway.ImmutablePayload;
+import com.github.princesslana.eriscasper.gateway.OpCode;
+import com.github.princesslana.eriscasper.gateway.Payload;
+import io.reactivex.annotations.Nullable;
+import org.immutables.value.Value;
+
+/**
+ * @see <a href="https://discordapp.com/developers/docs/topics/gateway#update-voice-state">
+ *     https://discordapp.com/developers/docs/topics/gateway#update-voice-state</a>
+ */
+@Value.Immutable
+public interface UpdateVoiceState extends GatewayCommand {
+  @JsonProperty("guild_id")
+  Snowflake getGuildId();
+
+  @JsonProperty("channel_id")
+  @Nullable
+  Snowflake getChannelId();
+
+  @JsonProperty("self_mute")
+  Boolean isMuted();
+
+  @JsonProperty("self_deaf")
+  Boolean isDeaf();
+
+  @Override
+  default Payload toPayload(ObjectMapper jackson) {
+    return ImmutablePayload.builder()
+        .op(OpCode.VOICE_STATE_UPDATE)
+        .d(jackson.valueToTree(this))
+        .build();
+  }
+
+  static UpdateVoiceState disconnect(Snowflake guildId) {
+    return connect(guildId, null);
+  }
+
+  static UpdateVoiceState connect(Snowflake guildId, Snowflake channelId) {
+    return connect(guildId, channelId, false, false);
+  }
+
+  static UpdateVoiceState connect(
```

Comment:
I think this starts to get too much. I think the client should just use the builder themselves instead of a four parameter method. Method signatures like (Snowflake, Snowflake, boolean, boolean) are error prone also. It's easy to accidentally flip the order of parameters.

Reply:
Gotcha, I'll remove the connect/disconnect methods, I thought it might be too much as well.